pycargoebuild/cargo.py

```diff
@@ -87,14 +105,20 @@ def get_package_metadata(f: typing.BinaryIO) -> PackageMetadata:
             "pycargoebuild on one of its members instead: "
             f"{' '.join(cargo_toml['workspace']['members'])}")
 
+    workspace_pkg_meta = cargo_toml.get("workspace", {}).get("package", {})
     pkg_meta = cargo_toml["package"]
-    pkg_license = pkg_meta.get("license")
+    _get_meta_key = functools.partial(get_meta_key,
+                                      pkg_meta=pkg_meta,
+                                      workspace_pkg_meta=workspace_pkg_meta)
+
+    pkg_license = _get_meta_key("license")
     if pkg_license is not None:
         pkg_license = cargo_to_spdx(pkg_license)
+
     return PackageMetadata(
         name=pkg_meta["name"],
-        version=pkg_meta["version"],
+        version=_get_meta_key("version", required=True),  # type: ignore
```

Comment:
Since this is the only required key, perhaps it would be cleaner to remove `required` logic from `get_meta_key()` and instead pull it as a var and check for `None` explicitly, then `# type: ignore` probably wouldn't be necessary.